torchgeo/datasets/biomassters.py

```diff
@@ -0,0 +1,344 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+"""BioMassters Dataset."""
+
+import os
+from typing import Any, Optional
+
+import matplotlib.pyplot as plt
+import numpy as np
+import rasterio
+import torch
+from matplotlib.figure import Figure
+from torch import Tensor
+
+from .geo import NonGeoDataset
+from .utils import download_url, extract_archive, percentile_normalization
+
+
+class BioMassters(NonGeoDataset):
+    """BioMassters Dataset for Aboveground Biomass prediction.
+
+    Dataset intented for Aboveground Biomass (AGB) prediction
+    over Finnish forests based on Sentinel 1 and 2 data with
+    corresponding target AGB mask values generated by Light Detection
+    and Ranging (LiDAR).
+
+    Dataset Format:
+
+    * .tif files for Sentinel 1 and 2 data
+    * .tif file for pixel wise AGB target mask
+    * .csv files for metadata regarding features and targets
+
+    Dataset Features:
+
+    * 13,000 target AGB masks of size (256x256px)
+    * 12 months of data per target mask
+    * Sentinel 1 and Sentinel 2 data for each location
+    * Sentinel 1 (specify channels here) available for each month
+    * Sentinel 2 (B02, B03, B04, B05, B06, B07, B08, B8A, B11, B12)
+      not available for each month due to ESA aquisition halt over the region
+      during particular periods
+
+    If you use this dataset in your research, please cite the following paper:
+
+    * TBD
```

Comment:
```suggestion
    * https://nascetti-a.github.io/BioMasster/
```
This is the closest thing we have for now, still citable even if it isn't on arXiv yet.